engine/internal/autoscaler/autoscaler.go

```diff
@@ -157,21 +163,25 @@ func (s *scaler) scale(ctx context.Context) error {
 		return client.IgnoreNotFound(err)
 	}
 
-	totalObserved := s.metricsClient.Get(s.modelID)
-	observedPerPods := totalObserved / float64(sts.Status.Replicas)
+	// TODO(aya): Stop scaling when there are too many not-ready pods
 
+	metricsVal := s.metricsClient.Get(s.modelID)
+	replicas := math.Min(float64(sts.Status.ReadyReplicas), float64(sts.Status.Replicas))
```

Comment:
might make sense to define our own `min` and `max` and avoid conversion to `float64`.